# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: testing/web-platform/tests/html/syntax/speculative-charset/support/script.py
# Commit: 9a8abd87cc79
# Full Hash: 9a8abd87cc7935f29b94248c1a6f8203faa14403
# Author: Henri Sivonen <hsivonen@hsivonen.fi>
# Date: 2021-12-08 21:43:06
# Regressor Bug: 1701828
# File Overlap Count: 10
# Description:
#   Bug 1701828 - meta charset rewrite. r=smaug
#   
#   Implements https://github.com/whatwg/html/issues/6962 . Improves performance
#   when <meta charset> occurs in head but after the first kilobyte and aligns
#   behavior better with WebKit and Blink.
# ==============================================================================

diff --git a/testing/web-platform/tests/html/syntax/speculative-charset/support/script.py b/testing/web-platform/tests/html/syntax/speculative-charset/support/script.py
new file mode 100644
--- /dev/null
+++ b/testing/web-platform/tests/html/syntax/speculative-charset/support/script.py
@@ -0,0 +1,18 @@
+import time
+
+def main(request, response):
+    response.add_required_headers = False # Don't implicitly add HTTP headers
+    response.writer.write_status(200)
+    response.writer.write_header("Content-Type", "text/javascript")
+    response.writer.end_headers()
+
+    token = request.GET[b"uuid"]
+    character = request.GET[b"character"]
+    old_character = b"NOT LOADED PREVIOUSLY";
+    from_stash = request.server.stash.take(token);
+    if from_stash:
+        old_character = from_stash
+    else:
+        request.server.stash.put(token, character)
+
+    response.writer.write(b'parent.postMessage("token: %s, character: %s, previous character: %s, byte: \xE6", "*");' % (token, character, old_character));